sklearn/utils/estimator_checks.py

```diff
@@ -832,25 +842,43 @@ def check_sample_weights_invariance(name, estimator_orig):
         X = np.array([[1, 3], [1, 3], [1, 3], [1, 3],
                       [2, 1], [2, 1], [2, 1], [2, 1],
                       [3, 3], [3, 3], [3, 3], [3, 3],
-                      [4, 1], [4, 1], [4, 1], [4, 1]], dtype=np.dtype('float'))
+                      [4, 1], [4, 1], [4, 1], [4, 1]], dtype=np.float64)
         y = np.array([1, 1, 1, 1, 2, 2, 2, 2,
-                      1, 1, 1, 1, 2, 2, 2, 2], dtype=np.dtype('int'))
+                      1, 1, 1, 1, 2, 2, 2, 2], dtype=np.int)
+
+        if kind == 'ones':
+            X2 = X
+            y2 = y
+            sw2 = None
+            err_msg = (f"For {name} sample_weight=None is not equivalent to "
+                       f"sample_weight=ones")
+        elif kind == 'zeros':
+            X2 = np.vstack([X, X])
```

Comment:
Deserves a comment like "Construct a dataset that is very different to (X, y) if weights are disregarded, but identical to (X, y) given weights".

However, this doesn't really work when the estimator is unsupervised. This could pass if weights are disregarded.